metrics/isolator.cpp

```diff
@@ -0,0 +1,580 @@
+#include <map>
+#include <string>
+#include <vector>
+
+#include <mesos/http.hpp>
+#include <mesos/mesos.hpp>
+#include <mesos/module.hpp>
+
+#include <mesos/module/isolator.hpp>
+#include <mesos/module/module.hpp>
+
+#include <process/address.hpp>
+#include <process/collect.hpp>
+#include <process/defer.hpp>
+#include <process/future.hpp>
+#include <process/process.hpp>
+
+#include <stout/assert.hpp>
+#include <stout/hashset.hpp>
+#include <stout/ip.hpp>
+#include <stout/nothing.hpp>
+#include <stout/numify.hpp>
+#include <stout/option.hpp>
+#include <stout/os.hpp>
+#include <stout/path.hpp>
+#include <stout/protobuf.hpp>
+#include <stout/strings.hpp>
+#include <stout/try.hpp>
+
+#include "isolator.hpp"
+
+#include "metrics/messages.pb.h"
+
+namespace inet = process::network::inet;
+namespace unix = process::network::unix;
+
+using namespace mesos;
+using namespace process;
+
+using std::map;
+using std::string;
+using std::vector;
+
+using mesos::slave::ContainerClass;
+using mesos::slave::ContainerLaunchInfo;
+using mesos::slave::ContainerState;
+using mesos::slave::Isolator;
+
+using mesos::modules::metrics::ContainerStartRequest;
+using mesos::modules::metrics::ContainerStartResponse;
+using mesos::modules::metrics::LegacyState;
+
+namespace mesosphere {
+namespace dcos {
+namespace metrics {
+
+namespace internal {
+
+// TODO(greggomann): Make use of `mesos::internal::serialize` when we get
+// access to 'common/http.hpp' from the Mesos codebase.
+string serialize(
```

Comment:
It doesn't seem like the other content types (non-JSON) are used anywhere.  Is there a plan or need to support other content types?